### PR TITLE
display delete recommendation banners logic for SPO ACO

### PR DESCRIPTION
**Pull request by xebu**

### CODE REVIEW

File: server/views/pages/caseSummary.njk [hunk 1]
```diff
@@ -68,6 +68,23 @@
...
                 </div>
             </div>
+            {% if flags.flagCancelRecommendationProcess and  recommendationBanner.display %}
+                
+                {% set bannerHtml %}
+                    <p>{{ recommendationBanner.createdByUserFullName }} {{ recommendationBanner.text}} {{ recommendationBanner.personOnProbationName }} on {{ formatDateTimeFromIsoString({ isoDate: recommendationBanner.createdDate, dateOnly: true }) }}.</p>
+                    <p>
+                        <a href="">Cancel</a> if you need to stop this recommendation. You will be asked why.
+                    </p>
+                    </form>
```
Comment (DarrenW-HMPPS) [inline server/views/pages/caseSummary.njk:78]: Looks like this form closing tag doesn't have an opening tag

**Review by DarrenW-HMPPS — CHANGES_REQUESTED** [review]

File: api/responses/get-recommendation-response.json [hunk 2]
```diff
@@ -0,0 +1,249 @@
+{
```
[the remark this answers is not in the document]
Comment (xebu) [reply api/responses/get-recommendation-response.json:1]: I would prefer to go with your suggestion, only brought in this file to pass the broken unit tests. I would like to have the `createdByFullName` in the response anyways. Did not wish to edit an existing file without first checking with the team

File: server/controllers/caseSummary/caseSummaryController.ts [hunk 3]
```diff
@@ -89,6 +112,17 @@ async function get(req: Request, res: Response, _: NextFunction) {
 
       const isAcoSignatureRequested = statuses.find(status => status.name === STATUSES.ACO_SIGNATURE_REQUESTED)
 
+      const isDoNotRecall = statuses.find(status => status.name === STATUSES.NO_RECALL_DECIDED)
+
+      const isRecallDecided = statuses.find(status => status.name === STATUSES.RECALL_DECIDED)
+
+      if (isRecommendationActive && !statuses.length) recommendationBanner.text = 'started a recommendation for'
+
```
[the remark this answers is not in the document]
Comment (xebu) [reply server/controllers/caseSummary/caseSummaryController.ts:120]: I would prefer a status for this, but did not know if it was possible at this stage.

File: server/middleware/featureFlags.ts [hunk 4]
```diff
@@ -25,6 +25,11 @@ export const featureFlagsDefaults: Record<string, FeatureFlagDefault> = {
     description:
       'Development team use only - shows links on the Recommendations tab allowing any recommendation to be marked as deleted. Deleting a recommendation allows a new one to be created, if needed. The "deleted" recommendation will be retained in the database, and no data or audit info will be lost.',
     default: false,
+  }, // TODO: set flagCancelRecommendationProcess to FALSE
```
[the remark this answers is not in the document]
Comment (xebu) [reply server/middleware/featureFlags.ts:28]: forgot to remove. `todo` removed.

File: server/controllers/caseSummary/caseSummaryController.ts [hunk 5]
```diff
@@ -26,6 +28,16 @@ interface RecommendationButton {
   link?: string
 }
 
+interface RecommendationBanner {
+  display: boolean
+  createdByUserFullName?: string
+  createdDate?: string
+  personOnProbationName?: string
+  dataAnalyticsEventCategory?: string // TODO: Confirm if this is required
```
[the remark this answers is not in the document]
Comment (xebu) [reply server/controllers/caseSummary/caseSummaryController.ts:36]: Agreed that Jatin to defined event as part of next code push.

File: server/views/pages/caseSummary.njk (hunk 1 above)
Comment (xebu) [reply server/views/pages/caseSummary.njk:78]: Removed

File: api/responses/get-recommendation-response.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (xebu) [reply api/responses/get-recommendation-response.json]: I have removed `get-recommendation-reponse.json`

File: server/controllers/caseSummary/caseSummaryController.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (xebu) [reply server/controllers/caseSummary/caseSummaryController.ts]: Added values for `dataAnalyticsEventCategory` with status either `NO_RECALL_DECIDED` or `RECALL_DECIDED` or `PO_START_RECALL`

File: server/controllers/caseSummary/caseSummaryController.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (xebu) [reply server/controllers/caseSummary/caseSummaryController.ts]: as above

File: server/controllers/caseSummary/caseSummaryController.ts [hunk 6]
```diff
@@ -26,6 +28,16 @@ interface RecommendationButton {
   link?: string
 }
 
+interface RecommendationBanner {
+  display: boolean
+  createdByUserFullName?: string
```
[the remark this answers is not in the document]
Comment (DarrenW-HMPPS) [reply server/controllers/caseSummary/caseSummaryController.ts:33]: But do the properties in this interface need to be nullable?  Probably not.  Only using nullables where they are actually required makes the software more robust.

[the remark this answers is not in the document]
Comment (xebu) [reply server/controllers/caseSummary/caseSummaryController.ts:33]: Yes to `nullable` as the `RecommendationBanner` info is fully populated for SPO / ACO and not PO. Follows convention of `RecommendationButton`. 

Is that ok?

**Review by DarrenW-HMPPS — APPROVED** [review]

**Pull request merged**